packages/@lightningjs/ui-components/src/mixins/withThemeStyles/StyleManager.js

```diff
@@ -0,0 +1,295 @@
+    this._boundThemeUpdate = null;
+    this.component = null;
+  }
+
+  /**
+   * Handler for the themeUpdate event. Clears the cache and updates the component's style.
+   * @private
+   */
+  _onThemeUpdate() {
+    // Why does this run so much
+    window.LUI_STYLE_CACHE && window.LUI_STYLE_CACHE.clear();
+    this.update();
+  }
+
+  /**
+   * Clears the source cache.
+   */
+  clearSourceCache() {
+    this._removeCache('styleSource');
+  }
+
+  /**
+   * Clears the style cache.
+   */
+  clearStyleCache() {
+    const { mode, tone } = this.component;
+    this._removeCache(`style_${mode}_${tone}`);
+  }
+
+  /**
+   * Generates a cache key.
+   * @param {string} name - The name of the cache.
+   * @returns {string} - The generated cache key.
+   */
+  _generateCacheKey(name) {
+    const cacheKey = [
+      name,
+      this.component.constructor.__componentName,
+      this._hasComponentStyle ? this.component.__id : null
+    ]
+      .filter(Boolean)
+      .join('_');
+    return cacheKey;
+  }
+
+  /**
+   * Adds a cache.
+   * @param {string} name - The name of the cache.
+   * @param {object} payload - The payload of the cache.
+   */
+  _addCache(name, payload) {
+    const key = this._generateCacheKey(name);
+    const existing = window.LUI_STYLE_CACHE.get(key);
+    window.LUI_STYLE_CACHE.set(key, {
+      ids: [...new Set([...(existing?.ids || []), this.component.__id])], // Add current id and remove duplicates
+      payload
+    });
+  }
+
+  /**
+   * Called when component is destroyed
+   */
+  _cleanupCache() {
+    if (!this.component) return;
+    window.LUI_STYLE_CACHE.forEach(({ ids, payload }, name) => {
+      const removeIndex = ids && ids.length && ids.indexOf(this.component.__id);
+      if (removeIndex > -1 && ids.length > 1) {
+        window.LUI_STYLE_CACHE.set(name, {
+          ids: ids.slice(0, removeIndex).concat(ids.slice(removeIndex + 1)),
+          payload
+        });
+      } else if (removeIndex > -1) {
+        window.LUI_STYLE_CACHE.delete(name);
+      }
+    });
+  }
+
+  /**
+   * Removes a cache.
+   * @param {string} name - The name of the cache.
+   */
+  _removeCache(name) {
+    const key = this._generateCacheKey(name);
+    window.LUI_STYLE_CACHE.delete(key);
+  }
+
+  /**
+   * Gets a cache.
+   * @param {string} name - The name of the cache.
+   * @returns {object|boolean} - The cache or false if the component has inline styles.
+   */
+  _getCache(name) {
+    const key = this._generateCacheKey(name);
+    return window.LUI_STYLE_CACHE.get(key);
+  }
+
+  /**
+   * Updates the component's style based on changes in tone, mode, theme, and `_componentLevelStyle`.
+   * Clears the style cache if tone or mode have changed, and clears the source cache if `_componentLevelStyle` has changed.
+   * Generates the component's style source and style objects, saves them to the cache, and emits a `styleUpdate` event with the updated style.
+   * @returns {Promise<void>}
+   */
+  async _update() {
```

Comment:
why does this need to be async?

Reply:
Was trying to keep it async incase we ever found value in processing styles with web workers in the future.